src/server/systems/NetworkingServer.cpp

```diff
@@ -194,9 +218,38 @@ void NetworkingServer::setClientUnreliableUpdate(sf::Packet packet, int client_i
                 m_playerUnreliableUpdates_lock);
             m_playerUnreliableUpdates_t.push_back(player_unreliable_input);
         }
-    } else {
-        std::cout << "Receive client_update command with tick drifted past drift threshold. "
-                  << "Drift value is: " << drift << std::endl;
+    }
+}
+
+
+float NetworkingServer::getBroadcastGameStateRate() {
+    {
+        std::lock_guard<std::mutex> m_playerIdsToUpdatesMetric_guard(m_playerIdsToUpdatesMetric_lock);
+        return m_broadcastGameStateMetric_t.getRate(sf::seconds(1));
+    }
+}
+
+std::unordered_map<sf::Uint32, float> NetworkingServer::getPlayerUnreliableUpdatesRates() {
+    {
+        std::lock_guard<std::mutex> m_playerIdsToUpdatesMetric_guard(m_playerIdsToUpdatesMetric_lock);
+        auto map = m_playerIdsToUpdatesMetric_t;
+        for (auto it = map.begin(); it != map.end(); ++it) {
+            m_playerIdsToUpdatesRates_t[it->first] = (it->second)->getRate(sf::seconds(1));
+        }
+
+        return m_playerIdsToUpdatesRates_t;
+    }
+}
+
+std::unordered_map<sf::Uint32, float> NetworkingServer::getPlayerTickDrifts() {
+    {
+        std::lock_guard<std::mutex> m_playerIdsToDriftMetric_guard(m_playerIdsToDriftMetrics_lock);
+        auto map = m_playerIdsToDriftMetrics_t;
```

Comment:
You dont need to copy the map here since you're holding the lock for the entire scope anyways,
you can just loop over `m_playerIdsToDriftMetrics_t` and copy the data into `m_playerIdsToDrifts_t`